docs/usage.md

```diff
@@ -85,6 +124,36 @@ after 'last_task_you_want_to_run' 'patches:run'
 
 If you are using sidekiq and restarting the sidekiq process on the box as a part of the deploy process, please make sure that the patches run task runs after sidekiq restarts, otherwise there is no guarentee the tasks will run.
 
+## Downloading Files
+
+To download a remote file, simply call ```download_file(filename, destination)```. The file will be downloaded using the configured downloader and stored in the destination. The file path will be returned if the file was successfully downloaded.
+
+```
+class MyPatch < Patches::Base
+  def run
+    filepath = download_file(filename, destination)
+    # do something with the downloaded file
+  end
+end
+```
+
+You can also use a different downloader on the fly by passing it to the download method:
+
+```
+class MyDownloader
+  def download(filename, destination)
+    # your custom download logic here
+  end
+end
+
+class MyPatch < Patches::Base
+  def run
+    filepath = download_file(filename, destination, MyDownloader.new)
```

Comment:
I think the Neptune implementation also offered a block format that would delete the file after use. That's a great concept as downloading patch files without any thought to removing them potentially causes disk fills or security concerns.

Reply:
For this initial release my approach was to add the least code possible so we can start using it ASAP and get some feedback from it.

Yes Neptune do have a block implementation but I couldn't find it being used anywhere so decided to leave it out. We can always add it later.

Still, if you think it's critical to have this from day one I can add it.

Reply:
I'd say add it. Every Gem release here, and update on the app side, requires coordination.